### DISCUSSION TITLE
Welcome to 42_GradeMe Discussions!

### DISCUSSION THREAD

**Discussion by JCluzet** [JCluzet/42_GradeMe#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nick-ras** [comment]
hi, is it possible to ask for a specific exervise for exam02?

**Comment by budindepunk** [comment]
Hey. In one exercise in Exam Rank 2, level 2, there is a problem. The exercise is `print_bits`. In the subject it says the function should take a char and print it in binary. but the example provided is wrong:

> 
"Example, if you pass 2 to print_bits, it will print "00000010""

00000010 is the binary of the NUMBER 2, not the CHAR 2. When passing 2 to the function it should print 00110010, which is 50, which is the ASCII value of the char '2'. This is the solution that passes the exam. The wording in the subject might be confusing, because if someone changes their code to print 00000010 they will fail.

**Reply by JCluzet** [reply]
Hey @budindepunk,

Thanks a lot for using Grademe :) I hope 42 is going well.

I want to clarify the confusion regarding the “print_bits” exercise.

**Type of data unsigned char:**

	- An unsigned char is an unsigned integer type, representing values from 0 to 255.
	- In the context of the exercise, it is meant to handle this value as a byte, not as an ASCII character.

**Provided example:**

	- Passing the value 2 to the print_bits function means passing the byte with the numeric value 2.
	- In binary, the value 2 is represented as 00000010, which is correct according to the provided example.

The confusion might stem from the fact that char is often used to represent ASCII characters. However, in this exercise, **unsigned char is treated as a direct numeric value**.

In summary, the exercise description is correct:

	- Passing the value 2 (as an unsigned char) to print_bits should indeed print “00000010”.

Thank you for your understanding.

**Reply by budindepunk** [reply]
You mean passing just 2, not '2' as a char? I tested my function passing '2', like so:
char c = '2';
print_bits(c);
and the output was 00110010. This passed. I did not test passing 2, without '  '. I do now with my same solution and indeed it prints 00000010.
I can see the phrasing causing some trouble for someone who might be passing '2' and getting what looks like a "wrong" result, when it isn't.
Thanks for your answer!

**Reply by JCluzet** [reply]
This passed, you mean using Grademe or at 42 exam?
